### DISCUSSION TITLE
Compatibility Problems with loading Relation Extraction Component

### DISCUSSION THREAD

**Discussion by ivankrstev7** [explosion/spaCy#11862] [Help: Coding & Implementations] [answered]
Hi.
I have trained several versions of relation extraction models, and from the last one I have problems loading it with `spacy.load()`. It throws me this exception:
![image](https://user-images.githubusercontent.com/48191509/203508227-a7fe0dfe-18dc-4456-8592-9f4ef5d570f9.png)
Something that solves the compatibility problem is installing spacy_transformers as:
`!pip3 install git+https://github.com/explosion/spacy-transformers`
But I need a stable version for production. Any ideas on how to avoid this issue?

**Comment by adrianeboyd** [comment, marked as answer]
Is it possible that you trained your relation extraction model with a source install of `spacy-transformers` rather than the last release (v1.1.8)?

**Reply by ivankrstev7** [reply]
Just checked the version from the training script and it is 1.0.4.
I guess retraining would be a possible solution.

**Reply by adrianeboyd** [reply]
The internal format did change a lot between v1.0 and v1.1. It should be forward compatible, so you should be able to load a model that was trained with v1.0 with v1.1, but once it's been saved with v1.1, it's converted to the new format and can't be loaded with v1.0 anymore. If it's being converted from the v1.0 format, you'll see this warning:

```
UserWarning: Automatically converting a transformer component from spacy-transformers v1.0 to v1.1+. If you see errors or degraded performance, download a newer compatible model or retrain your custom model with the current spacy-transformers version. For more details and available updates, run: python -m spacy validate
```

The error above looks to me more like you trained and saved with a source install of `spacy-transformers` (at some point after https://github.com/explosion/spacy-transformers/pull/339) and are trying to reload with an earlier version.
